### PR TITLE
chore!: Make constructor private

### PR DESCRIPTION
**Pull request by FrankEssenberger**
In a [support issuse](https://answers.sap.com/questions/13764308/sf-api-select-using-navigation-property.html) a customer used the constructor to get a API object. This is not good, since the init step is not invoked. 

This PR makes the constructor private to protect users (at least if they use TypeScript or a IDE which understands the private keyword).

The static method I introduced is called `_privateFactory` - here I am up for any good suggestion.

### CODE REVIEW

**Review by fwilhe — APPROVED** [review]

File: .changeset/loud-radios-arrive.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sap-cloud-sdk/odata-common': patch
```
Comment (jjtang1985) [inline .changeset/loud-radios-arrive.md:2]:
```suggestion
'@sap-cloud-sdk/odata-common': major
```

File: .changeset/loud-radios-arrive.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+[Compatibility Note] The constructor of the entity API is private to avoid accidental usage. Use the service object to get a API instance.
```
Comment (jjtang1985) [inline .changeset/loud-radios-arrive.md:5]: [q] Should we update the "upgrade guide"?

File: packages/generator/src/generator-without-ts-morph/entity-api/class.ts [hunk 3]
```diff
@@ -25,10 +25,18 @@ export function classContent(
...
   public deSerializers: DeSerializersT;
 
-  constructor(
+  private constructor(
     deSerializers: DeSerializersT = defaultDeSerializers as any) {
     this.deSerializers = deSerializers;
   }
+  
+  ${documentationBlock`Do not use this method or the constructor directly.
+Use the service object to get a API instance.`}
```
Comment (jjtang1985) [inline packages/generator/src/generator-without-ts-morph/entity-api/class.ts:34]:
```suggestion
Use the service object to get an API instance.`}  
```

**Review by jjtang1985 — COMMENTED** [review]
Looks good.
As this is not refined, maybe ask for more review.
I left some comments.

Other alternatives:
- lint rule for checking the constructor use case (lint and the rules are required)
- runtime error that shows the wrong usage (over engineering?)

**Pull request merged**